### PR TITLE
[PA-1233] Add StartPxBackupTorpedoTest and EndPxBackupTorpedoTest

### PR DESCRIPTION
**Pull request by kphalgun-px**
**What this PR does / why we need it**:
This PR adds new StartPxBackupTorpedoTest and EndPxBackupTorpedoTest to assist in dealing with controllers and automating the automation tracker.

**Which issue(s) this PR fixes** (optional)
Closes #PA-1233

**Special notes for your reviewer**:
Please review the Jenkins build for the "BasicBackupCreation" test case using the link below

https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/Custom-Pipelines/job/px-backup-on-demand-system-test-byoc/1603/

Aetos Dashboard: https://aetos.pwx.purestorage.com/resultSet/testSetID/215701

### CODE REVIEW

File: tests/backup/backup_test.go
Comment (kphalgun-px) [inline tests/backup/backup_test.go:1]: I will undo the changes before merging.

File: drivers/backup/helper/backup_helper.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	return nil
+}
+
+// EndPxBackupTorpedoTest ends the specified test and performs cleanup
+func EndPxBackupTorpedoTest(testRailId int) error {
+	if pxBackupTorpedoTestInfo, ok := pxBackupTorpedoTestInfoMap[testRailId]; ok {
+		tests.CloseLogger(pxBackupTorpedoTestInfo.testLogger)
+		tests.Inst().Dash.TestCaseEnd()
+		if tests.TestRailSetupSuccessful && pxBackupTorpedoTestInfo.testRailID != 0 && pxBackupTorpedoTestInfo.testRunIdForSuite != 0 {
+			contexts := make([]*scheduler.Context, 0)
```
Comment (kphalgun-px) [inline drivers/backup/helper/backup_helper.go:82]: I will address this issue in PR #1414.

**Review by snigdha-px — APPROVED** [review]
LGTM

File: drivers/backup/helper/backup_helper.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	testDescription   string
+	testAuthor        string
+	testRailID        int
+	testRunIdForSuite int
+	testTags          map[string]string
+	testLogger        *lumberjack.Logger
+}
+
+// StartPxBackupTorpedoTest creates a logger, configures the Aetos Dashboard for the specified test, and initializes controllers
+func StartPxBackupTorpedoTest(testRailId int, testName string, testDescription string, testAuthor string, apps []string, tags ...map[string]string) error {
```
Comment (mkoppal-px) [inline drivers/backup/helper/backup_helper.go:30]: For some tests like `CreateMultipleUsersAndGroups` there are no `apps` deployed. How do we handle that?

File: drivers/backup/helper/backup_helper.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+		}
+		for tagKey, tagValue := range tags[0] {
+			for _, invalidTagKey := range invalidTagKeys {
+				if tagKey == invalidTagKey {
+					err := fmt.Errorf("tags cannot have any of the keys [%s]", invalidTagKeys)
+					return err
+				}
+			}
+			testTags[tagKey] = tagValue
+		}
```
Comment (mkoppal-px) [inline drivers/backup/helper/backup_helper.go:54]: Why is this needed?

File: drivers/backup/helper/backup_helper.go (hunk 2 above)
Comment (kphalgun-px) [reply drivers/backup/helper/backup_helper.go:30]: Pass empty array or nil.

File: drivers/backup/helper/backup_helper.go (hunk 3 above)
Comment (kphalgun-px) [reply drivers/backup/helper/backup_helper.go:54]: To prevent from overriding the author and apps keys.

Comment (mkoppal-px) [reply drivers/backup/helper/backup_helper.go:54]: This is not required

File: drivers/backup/helper/backup_helper.go (hunk 2 above)
Comment (mkoppal-px) [reply drivers/backup/helper/backup_helper.go:30]: This may fail the API call to aetos if we have nil value

Comment (mkoppal-px) [reply drivers/backup/helper/backup_helper.go:30]: Can you test this?

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Review by mkoppal-px — APPROVED** [review]
LGTM

File: drivers/backup/helper/backup_helper.go (hunk 2 above)
Comment (kphalgun-px) [reply drivers/backup/helper/backup_helper.go:30]: https://go.dev/play/p/0q1VnWrKgqa
